python/lsst/ts/criopy/m1m3/BumpTestTimes.py

```diff
@@ -0,0 +1,112 @@
+# this program. If not, see <https://www.gnu.org/licenses/>.
+
+from astropy.time import Time, TimeDelta
+from lsst_efd_client import EfdClient
+
+from ..M1M3FATable import FAType, force_actuator_from_id
+
+
+class BumpTestTimes:
+
+    """Returns a set of times for bump tests given the actuator ID and
+    an input timespan
+
+    Parameters
+    ----------
+    client : lsst_efd_client.efd_helper.EfdClient object
+        This is used to query the EFD
+    """
+
+    def __init__(self, client: EfdClient = EfdClient("summit_efd")):
+        super().__init__()
+        self.client = client
+
+    async def find_times(
+        self,
+        actuator_id: int,
+        start: Time = (Time.now() - TimeDelta(7, format="jd")),
+        end: Time = Time.now(),
+    ) -> tuple[list[float], list[float]]:
+        """Find bump test query times
+        actuator_id : `int`
+            Force Actuator identification number. Starting with 101, the first
+            number identified segment (1-4). The value ranges up to 443.
+
+        start: 'Time'
+            Astropy Time of search start.
+
+        end: Time
+            Astropy Time of search end.
+
+        returns: Two nested list of ints, one for the primary bump and one for
+            the secondary bump [start,end], [start, end]...]
+            These times will be in unix_tai format
+        """
+        fa = force_actuator_from_id(actuator_id)
+        manyBumps = await self.client.select_time_series(
+            "lsst.sal.MTM1M3.logevent_forceActuatorBumpTestStatus", "*", start, end
+        )
+
+        theseBumps = manyBumps[manyBumps["actuatorId"] == actuator_id]
+        # Find the test names
+        primaryBump = f"primaryTest{fa.index}"
+        if fa.actuator_type == FAType.DAA:
+            secondaryBump = f"secondaryTest{fa.s_index}"
+        else:
+            secondaryBump = None
+        # Now find the separate tests
+        times = theseBumps["timestamp"].values
+        startTimes = []
+        endTimes = []
+        for i, time in enumerate(times):
+            if i == 0:
+                startTimes.append(time)
+                continue
+            if (time - times[i - 1]) > 60.0:
+                startTimes.append(time)
+                endTimes.append(times[i - 1])
+        endTimes.append(times[-1])
+        # Now use these to find the bump test start and end times
+        primaryTimes: list[tuple[float, float]] = []
+        secondaryTimes: list[tuple[float]] = []
+        for startTime, endTime in zip(startTimes, endTimes):
+            thisBump = theseBumps[
+                (theseBumps["timestamp"] >= startTime)
+                & (theseBumps["timestamp"] <= endTime)
+            ]
+            try:
+                plotStart = (
+                    thisBump[thisBump[primaryBump] == 2]["timestamp"].values[0] - 1.0
+                )
+                plotEnd = plotStart + 14.0
+                primaryTimes.append([plotStart, plotEnd])
```

Comment:
These are always fixed lengths, consider using tuples instead of lists, e.g.;

Here and elsewhere:

```
                primaryTimes.append((plotStart, plotEnd))
```

Then you could update the docstring to reflect the fact you are now returning a list of tuples.